### PR TITLE
feat: add sso login establish session usecase [AR-1354] [AR-1156] [AR-1157]

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Add delete and hide message

### Issues
[AR-1354]
[AR-1156]
[AR-1157]

### Testing
WIP
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[AR-1354]: https://wearezeta.atlassian.net/browse/AR-1354
[AR-1156]: https://wearezeta.atlassian.net/browse/AR-1156
[AR-1157]: https://wearezeta.atlassian.net/browse/AR-1157

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/auth/login/SSOLoginRepository.kt [hunk 1]
```diff
@@ -21,6 +22,8 @@ interface SSOLoginRepository {
 
     suspend fun finalize(cookie: String, serverConfig: ServerConfig): Either<NetworkFailure, String>
 
+    suspend fun ssoEstablishSession(cookie: String, serverConfig: ServerConfig): Either<NetworkFailure, SessionDTO>
```
Comment (saleniuk) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/auth/login/SSOLoginRepository.kt:25]: Do we need the `sso` prefix? I think that if we're using `SSOLoginRepository` that's saying enough that it's being used for sso, at least for me 😄

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/sso/SSOUtil.kt [hunk 2]
```diff
@@ -2,12 +2,15 @@ package com.wire.kalium.logic.data.sso
...
-    const val QUERY_USER_ID = "user"
+    const val QUERY_USER_ID = "userId"
+    const val QUERY_COOKIE = "cookie"
     const val QUERY_SERVER_CONFIG = "location"
-    const val SUCCESS_HOST = "sso-success.kalium"
-    const val ERROR_HOST = "sso-error.kalium"
+    const val QUERY_ERROR = "errorCode"
+    const val SSO_LOGIN_HOST = "sso-login"
+    const val SUCCESS_PATH = "success"
+    const val ERROR_PATH = "failure"
```
Comment (saleniuk) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/sso/SSOUtil.kt:15]: Maybe they should be private 🤔

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/user/login/SSOLoginApi.kt [hunk 3]
```diff
@@ -57,6 +66,31 @@ class SSOLoginApiImpl(private val httpClient: HttpClient) : SSOLoginApi {
         }
     }
 
+    override suspend fun ssoEstablishSession(cookie: String, apiBaseUrl: Url): NetworkResponse<SessionDTO> = wrapKaliumResponse<AccessTokenDTO> {
+            httpClient.post {
+                setUrl(apiBaseUrl, PATH_ACCESS)
```
Comment (saleniuk) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/user/login/SSOLoginApi.kt:71]: I'm thinking if it should be in the `SSOLoginApi` as it's not a request dedicated exclusively to the SSO 🤔

Comment (mchenani) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/user/login/SSOLoginApi.kt:71]: I get your point, but we already such a function with different logic in the login api and this one is specifically for the sso login.

**Review by saleniuk — APPROVED** [review]
🕺

**Pull request merged**